front/lib/api/oauth/providers/slack.ts

```diff
@@ -182,7 +231,29 @@ export class SlackOAuthProvider implements BaseOAuthStrategyProvider {
       useCase: OAuthUseCase;
     }
   ): Promise<ExtraConfigType> {
-    if (useCase === "personal_actions") {
+    if (useCase === "connection") {
+      // Remove the secrets from the stored config (they're stored in relatedCredential)
+      const {
+        slack_client_secret: _,
+        slack_signing_secret: __,
+        ...restConfig
+      } = extraConfig;
+
+      const featureFlags = await getFeatureFlags(
+        auth.getNonNullableWorkspace()
+      );
+      if (
+        featureFlags.includes("self_created_slack_app_connector_rollout") &&
+        restConfig.slack_client_id
+      ) {
+        return {
+          ...restConfig,
+          client_id: restConfig.slack_client_id,
```

Comment:
nit: we could have defined extra config as `client_id` | `client_secret` | `signing_secret` directly
It's what we do for other providers like salesforce